packages/repository/src/repositories/inclusion.ts

```diff
@@ -0,0 +1,96 @@
+import {Entity} from '../model';
+import {Getter} from '@loopback/core';
+import {
+  Filter,
+  Where,
+  resolveHasManyMetadata,
+  resolveBelongsToMetadata,
+  RelationMetadata,
+  RelationType,
+  constrainWhere,
+} from '../';
+import {AnyObject} from '..';
+import {DefaultCrudRepository} from './legacy-juggler-bridge';
+import {inspect} from 'util';
+import {
+  HasManyDefinition,
+  BelongsToDefinition,
+  HasManyResolvedDefinition,
+  BelongsToResolvedDefinition,
+} from '../relations';
+
+type ResolvedRelationMetadata =
+  | HasManyResolvedDefinition
+  | BelongsToResolvedDefinition;
+
+// SE: the source entity
+// TE: the target entity
+// SID: the ID of source entity
+// TID: the ID of target entity
+
+export class InclusionHandler<SE extends Entity, SID> {
+  _handlers: {[relation: string]: Function} = {};
+  constructor(public sourceRepository: DefaultCrudRepository<SE, SID>) {}
+
+  registerHandler<TE extends Entity, TID>(
+    relationName: string,
+    targetRepoGetter: Getter<DefaultCrudRepository<TE, TID>>,
+  ) {
+    this._handlers[relationName] = fetchIncludedItems;
+    const self = this;
+
+    async function fetchIncludedItems(
+      fks: SID[],
+      filter?: Filter<TE>,
+    ): Promise<TE[]> {
+      const targetRepo = await targetRepoGetter();
+      const relationDef: ResolvedRelationMetadata = self.getResolvedRelationDefinition(
+        relationName,
+      );
+      filter = filter || {};
+      filter.where = self.buildConstrainedWhere<TE>(
+        fks,
+        filter.where || {},
+        relationDef,
+      );
+      console.log(`inclusion filter: ${inspect(filter)}`);
+
+      return await targetRepo.find(filter);
+    }
+  }
+
+  findHandler(relationName: string) {
+    const errMsg =
+      `The inclusion handler for relation ${relationName} is not found!` +
+      `Make sure you defined ${relationName} properly.`;
+
+    return this._handlers[relationName] || new Error(errMsg);
```

Comment:
It's suspicious to return an `Error` but later on it checks:
```ts
const handler = this._inclusionHandler.findHandler(relation);
    if (!handler) {
      throw new Error('Fetch included items is not supported');
    }
```

Reply:
yeah the code here should definitely be refined ^